contracts/Staking.sol

```diff
@@ -482,6 +489,48 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         }
     }
 
+    function slashIndexer(address _indexer, uint256 _amount) external {
+        require(msg.sender == settings.getDisputeManager(), 'Only DisputeManager');
+        require(_amount <= this.getSlashableAmount(_indexer), 'slash over the slashable amount');
+
+        uint256 amount = _amount;
+
+        uint256 withdrawingLength = unbondingLength[_indexer] - withdrawnLength[_indexer];
+        if (withdrawingLength > 0) {
+            uint256 latestWithdrawnLength = withdrawnLength[_indexer];
+            for (uint256 i = latestWithdrawnLength; i < latestWithdrawnLength + withdrawingLength; i++) {
```

Comment:
Is it just `i = withdrawnLength[_indexer]; i < unbondingLength[_indexer]; i++` ?
And and remove variables `withwrawingLength` and `latestWithdrawnLength`, same as below `getSlashableAmount `